pkg/mock/phase_pending_test.go

```diff
@@ -0,0 +1,297 @@
+			},
+			Spec: v1.CycleNodeRequestSpec{
+				NodeGroupsList: []string{"ng-1"},
+				CycleSettings: v1.CycleSettings{
+					Concurrency: 1,
+					Method:      v1.CycleNodeRequestMethodDrain,
+				},
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"customer": "kitt",
+					},
+				},
+			},
+		}),
+		StartAtPhase(v1.CycleNodeRequestPending),
+		WithKubeNodes(nodegroup),
+	)
+
+	_, err = mockTransitioner.Run()
+	assert.Error(t, err)
+	assert.Equal(t, v1.CycleNodeRequestHealing, mockTransitioner.Cnr.Status.Phase)
+}
+
+// Test to ensure that if there's a mismatch between the instances found in the
+// cloud provider and kube then the CNR will error out immediately rather than
+// proceed. Specifically test missing kube nodes.
+func TestPendingNoKubeNodes(t *testing.T) {
+	nodegroup, err := newNodegroup("ng-1", 2)
+	if err != nil {
+		assert.NoError(t, err)
+	}
+
+	mockTransitioner := NewMockTransitioner(
+		WithCycleNodeRequest(&v1.CycleNodeRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cnr-1",
+				Namespace: "kube-system",
+			},
+			Spec: v1.CycleNodeRequestSpec{
+				NodeGroupsList: []string{"ng-1"},
+				CycleSettings: v1.CycleSettings{
+					Concurrency: 1,
+					Method:      v1.CycleNodeRequestMethodDrain,
+				},
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"customer": "kitt",
+					},
+				},
+			},
+		}),
+		StartAtPhase(v1.CycleNodeRequestPending),
+		WithCloudProviderInstances(nodegroup),
+	)
+
+	_, err = mockTransitioner.Run()
+	assert.Error(t, err)
+	assert.Equal(t, v1.CycleNodeRequestHealing, mockTransitioner.Cnr.Status.Phase)
+}
+
+// Test to ensure that Cyclops will not proceed if there is node detached from
+// the nodegroup on the cloud provider. It should try to wait for the issue to
+// resolve transition to the Healing phase if it doesn't.
+func TestPendingDetachedCloudProviderNode(t *testing.T) {
+	nodegroup, err := newNodegroup("ng-1", 2)
+	if err != nil {
+		assert.NoError(t, err)
+	}
+
+	// "detach" one instance from the asg
+	nodegroup[0].Nodegroup = ""
+
+	mockTransitioner := NewMockTransitioner(
+		WithCycleNodeRequest(&v1.CycleNodeRequest{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "cnr-1",
+				Namespace: "kube-system",
+			},
+			Spec: v1.CycleNodeRequestSpec{
+				NodeGroupsList: []string{"ng-1"},
+				CycleSettings: v1.CycleSettings{
+					Concurrency: 1,
+					Method:      v1.CycleNodeRequestMethodDrain,
+				},
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"customer": "kitt",
+					},
+				},
+			},
+		}),
+		StartAtPhase(v1.CycleNodeRequestPending),
+		WithKubeNodes(nodegroup),
+		WithCloudProviderInstances(nodegroup),
+	)
+
+	// Should requeue while it tries to wait
+	_, err = mockTransitioner.Run()
+	assert.NoError(t, err)
+	assert.Equal(t, v1.CycleNodeRequestPending, mockTransitioner.Cnr.Status.Phase)
+
+	// Simulate waiting for 1 day, this is more than the waiting limit
```

Comment:
Are we assuming the waiting limit is always < 24 hours? Can we make this an input into the test, so that someone changing the wait limit doesn't break unrelated tests